### PR TITLE
[FLINK-12440][python] Add all connector support align Java Table API.

### PR DESCRIPTION
**Pull request by WeiZhong94**
## What is the purpose of the change

This pull request add 2 `ConnectorDescriptor` classes for flink python API: `Kafka` and `Elasticsearch`. Because no appropriate format descriptor is available, we can't add end to end tests currently. So only the tests for descriptor APIs are added.

## Brief change log

  - *Add 2 `ConnectorDescriptor`: `Kafka` and `Elasticsearch`.*
  - *Adjust `travis_controller.sh` and `pyflink-gateway-server.sh` to include necessary jars.*
  - *Add tests for the new descriptors.*

## Verifying this change

This change added tests and can be verified as follows:

Added integration tests in `test_descriptorr.py`.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (yes)
  - If yes, how is the feature documented? (python docs)

### CODE REVIEW

File: flink-dist/src/main/flink-bin/bin/pyflink-gateway-server.sh [hunk 1]
```diff
@@ -63,7 +63,7 @@ if [[ -n "$FLINK_TESTING" ]]; then
     else
       FLINK_TEST_CLASSPATH="$FLINK_TEST_CLASSPATH":"$testJarFile"
     fi
-  done < <(find "$FLINK_SOURCE_ROOT_DIR" ! -type d -name 'flink-*-tests.jar' -print0 | sort -z)
+  done < <(find "$FLINK_SOURCE_ROOT_DIR" ! -type d \( -name 'flink-*-tests.jar' -o -path '*flink-connector-*-base/target/flink*.jar' \) -print0 | sort -z)
```
Comment (dianfu) [inline flink-dist/src/main/flink-bin/bin/pyflink-gateway-server.sh]: Specify flink-connector-kafka-base and flink-connector-elasticsearch-base explicitly as flink-connector-*-base is not a good pattern and these two connectors just happen to have this pattern.

File: flink-python/pyflink/java_gateway.py [hunk 2]
```diff
@@ -98,6 +98,7 @@ def preexec_func():
         gateway_parameters=GatewayParameters(port=gateway_port, auto_convert=True))
 
     # Import the classes used by PyFlink
+    java_import(gateway.jvm, "java.util.Properties")
```
Comment (dianfu) [inline flink-python/pyflink/java_gateway.py]: I'm suggesting to import java.util.Properties at the places they are used. We can follow the strategy to import the commonly used packages in advance and for the packages used seldom, we can put off to import them at the places they are used.

File: flink-python/pyflink/table/table_descriptor.py [hunk 3]
```diff
@@ -373,6 +375,485 @@ def path(self, path_str):
...
 
+class Kafka(ConnectorDescriptor):
+    """
+    Connector descriptor for the Apache Kafka message queue.
+    """
+
+    def __init__(self):
+        gateway = get_gateway()
+        self._j_kafka = gateway.jvm.Kafka()
+        super(ConnectorDescriptor, self).__init__(self._j_kafka)
```
Comment (dianfu) [inline flink-python/pyflink/table/table_descriptor.py]: super(Kafka, self)

File: tools/travis_controller.sh [hunk 4]
```diff
@@ -161,6 +161,7 @@ if [ $STAGE == "$STAGE_COMPILE" ]; then
             find "$CACHE_FLINK_DIR" -maxdepth 8 -type f -name '*.jar' \
             ! -path "$CACHE_FLINK_DIR/flink-dist/target/flink-*-bin/flink-*/lib/flink-dist*.jar" \
             ! -path "$CACHE_FLINK_DIR/flink-dist/target/flink-*-bin/flink-*/opt/flink-table*.jar" \
+            ! -path "$CACHE_FLINK_DIR/flink-connectors/*-base/target/flink-*.jar" \
```
Comment (dianfu) [inline tools/travis_controller.sh]: as commented above

**Comment by dianfu** [comment]
@WeiZhong94 Thanks a lot for the update. LGTM. +1 from my side.

**Comment by WeiZhong94** [comment]
@dianfu Thanks for your review! I have made changes according to your comments.

File: tools/travis_controller.sh (hunk 4 above)
Comment (WeiZhong94) [reply tools/travis_controller.sh]: Done.

File: flink-python/pyflink/table/table_descriptor.py (hunk 3 above)
Comment (WeiZhong94) [reply flink-python/pyflink/table/table_descriptor.py]: Done.

File: flink-python/pyflink/java_gateway.py (hunk 2 above)
Comment (WeiZhong94) [reply flink-python/pyflink/java_gateway.py]: Done.

File: flink-dist/src/main/flink-bin/bin/pyflink-gateway-server.sh (hunk 1 above)
Comment (WeiZhong94) [reply flink-dist/src/main/flink-bin/bin/pyflink-gateway-server.sh]: Done.

**Comment by dianfu** [comment]
@WeiZhong94 Thanks a lot for the update. LGTM. +1 from my side.

**Comment by sunjincheng121** [comment]
LGTM. +1 to merged.
@flinkbot approve all

**Pull request closed without merging**